clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -321,11 +325,15 @@ void runOnce() {
         }
 
         long currentTimeMs = time.milliseconds();
-        long pollTimeout = sendProducerData(currentTimeMs);
+        if (!dynamicConfig.shouldDisable()) {
+            dynamicConfig.maybeFetchConfigs(currentTimeMs);
```

Comment:
Two consecutive calls to `dynamicConfig` this can be done in a method for `dynamicConfig` and simplify the API.